Implement bitmap transcoding for most supported PixelFormats

Gillibald opened the pull request:
## What does the pull request do?
<!--- Give a bit of background on the PR here, together with links to with related issues etc. -->
This PR introduces a new interface (`IPixelFormatWriter`) for writing pixels of different formats.

The `IPixelFormatWriter` is then used in the new `PixelFormatTranscoder` class that allows the transcoding of pixels into a different format.

The previous helper method to transcode a bitmap was replaced by the new component.

### New API
`void Bitmap.CopyPixels(ILockedFramebuffer buffer, AlphaFormat alphaFormat)`

## What is the current behavior?
<!--- If the PR is a fix, describe the current incorrect behavior, otherwise delete this section. -->


## What is the updated/expected behavior with this PR?
<!--- Describe how to test the PR. -->


## How was the solution implemented (if it's not obvious)?
<!--- Include any information that might be of use to a reviewer here. -->


## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Breaking changes
<!--- List any breaking changes here. -->

## Obsoletions / Deprecations
<!--- Obsolete and Deprecated attributes on APIs MUST only be included when discussed with Core team. @grokys, @kekekeks & @danwalmsley -->

## Fixed issues
<!--- If the pull request fixes issue(s) list them like this: 
Fixes #123
Fixes #456
-->

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0039289-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0039307-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0039329-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0039363-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0039381-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

Gillibald committed:
Introduce PixelFormatWriter
Introduce PixelFormatTranscoder
Introduce Bitmap.CopyPixels that transcodes pixel and alpha format

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0039387-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]
